lib/src/configuration.dart

```diff
@@ -792,3 +850,84 @@ enum GeneralSyncErrorCode {
   final int code;
   const GeneralSyncErrorCode(this.code);
 }
+
+/// Contains the details for a compensating write performed by the server.
+/// {@category Sync}
+class CompensatingWriteInfo {
+  /// The type of the object which was affected by the compensating write.
+  final String objectType;
+
+  /// The reason for the server to perform a compensating write.
+  final String reason;
+
+  /// The primary key of the object which was affected by the compensating write.
+  final RealmValue primaryKey;
+
+  const CompensatingWriteInfo(this.objectType, this.reason, this.primaryKey);
+
+  @override
+  String toString() {
+    return "CompensatingWriteInfo: objectType: '$objectType' reason: '$reason' primaryKey: '$primaryKey'";
+  }
+}
+
+/// An error type that describes a compensating write error,
+/// which indicates that one more object changes have been reverted
+/// by the server.
+/// {@category Sync}
+class CompensatingWriteError extends SyncError {
+  /// The [CompensatingWriteError] has error code of [SyncSessionErrorCode.compensatingWrite]
+  SyncSessionErrorCode get code => SyncSessionErrorCode.compensatingWrite;
+
+  /// The list of the compensating writes performed by the server.
+  late final List<CompensatingWriteInfo>? compensatingWrites;
+
+  CompensatingWriteError._(
+    String message, {
+    String? detailedMessage,
+    this.compensatingWrites,
+  }) : super(message, SyncErrorCategory.session, SyncSessionErrorCode.compensatingWrite.code, detailedMessage: detailedMessage);
+
+  @override
+  String toString() {
+    return "CompensatingWriteError message: $message category: $category code: $code. ${compensatingWrites ?? ''}";
+  }
+}
+
+/// @nodoc
+extension SyncErrorInternal on SyncError {
+  static SyncError createSyncError(SyncErrorDetails error, {App? app}) {
+    if (error.isClientResetRequested) {
+      //Client reset can be requested with isClientResetRequested disregarding the SyncClientErrorCode and SyncSessionErrorCode values
+      return ClientResetError(error.message,
+          app: app,
+          category: error.category,
+          errorCodeValue: error.code,
+          originalFilePath: error.originalFilePath,
+          backupFilePath: error.backupFilePath,
+          detailedMessage: error.detailedMessage);
+    }
+
+    switch (error.category) {
+      case SyncErrorCategory.client:
```

Comment:
You are changing the behavior too much it seems. Previously we could create `ClientResetError` in this case as well.

Reply:
`[is_client_reset_requested](https://github.com/realm/realm-core/blob/41671b7661cb48d3c47ce974a9bec5e29595bce2/src/realm/sync/config.cpp#L79)` that is returned by the core will be true in case of having `Client::Error::auto_client_reset_failure`. That's why I removed this double check from here. It is not needed.

Reply:
it is fine by me if are really sure about that

Reply:
@desistefanova Seems sure this is correct so I am resolving